hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResource.java

```diff
@@ -263,6 +264,11 @@ public ResponseEntity<MgmtRolloutGroupResponseBody> getRolloutGroup(@PathVariabl
 
         final RolloutGroup rolloutGroup = rolloutGroupManagement.getWithDetailedStatus(groupId)
                 .orElseThrow(() -> new EntityNotFoundException(RolloutGroup.class, rolloutId));
+
+        if (!Objects.equals(rolloutId, rolloutGroup.getRollout().getId())) {
+            throw new ValidationException("The provided rolloutId does not match with the rolloutId in the deployment group.");
```

Comment:
As this is a GET request which addresses the group directly, I think we should rather return a 404 Not Found (instead of a 400 Bad Request).

Reply:
Applied some changes regarding this.